.github/workflows/asv-benchmarking-pr.yml

```diff
@@ -0,0 +1,107 @@
+name: ASV Benchmarking (PR)
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize, labeled]
+  workflow_dispatch:
+
+jobs:
+  benchmark:
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'run-benchmark') && github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch' }}
+    name: Linux
+    runs-on: ubuntu-latest
+    env:
+      ASV_DIR: "./benchmarks"
+      CONDA_ENV_FILE: ci/environment.yml
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+
+      - name: Set up Conda environment
+        uses: mamba-org/setup-micromamba@v1
+        with:
+          environment-file: ${{env.CONDA_ENV_FILE}}
+          cache-environment: true
+          environment-name: uxarray_build
+          cache-environment-key: "${{runner.os}}-${{runner.arch}}-py${{env.PYTHON_VERSION}}-${{env.TODAY}}-${{hashFiles(env.CONDA_ENV_FILE)}}-benchmark"
+          create-args: >-
+            asv
+            build
+            mamba
+
+      - name: Run Benchmarks
+        shell: bash -l {0}
+        id: benchmark
+        run: |
+          set -x
+          # ID this runner
+          asv machine --yes
+          echo "Baseline:  ${{ github.event.pull_request.base.sha }} (${{ github.event.pull_request.base.label }})"
+          echo "Contender: ${GITHUB_SHA} (${{ github.event.pull_request.head.label }})"
+          # Run benchmarks for current commit against base
+          ASV_OPTIONS="--split --show-stderr"
+          asv continuous $ASV_OPTIONS ${{ github.event.pull_request.base.sha }} ${GITHUB_SHA}
+          # Save compare results
+          asv compare --split ${{ github.event.pull_request.base.sha }} ${GITHUB_SHA} > asv_compare_results.txt
+        working-directory: ${{ env.ASV_DIR }}
+
+      - uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: asv-benchmark-results-${{ runner.os }}
+          path: |
+            ${{ env.ASV_DIR }}/results
```

Comment:
A few nice to have requests: 

1. Table broken down into various segments - mem, open time etc. 
2. Maintain another repo for keeping a record a results. Something where we can just based on commit sha track what changes led to the changes in performance.
3. A way to get the performance difference between two dates from the point number 2 mentioned above.

Reply:
> Table broken down into various segments - mem, open time etc.

This is something that we could do. There is a `--sort` flag that could sort the results based on name or ratio. 

> Maintain another repo for keeping a record a results. Something where we can just based on commit sha track what changes led to the changes in performance.

We already do this with our package benchmark (which is our other automated workflow). It keeps track of a record of results for each commit that is part of the main branch.

https://github.com/UXARRAY/uxarray-asv

For example, looking at the web page for one of the benchmarks, each of the points represents a run for a certain commit.

https://uxarray.github.io/uxarray-asv/#face_bounds.FaceBounds.peakmem_face_bounds

I don't think it's worth keeping the results from this PR, other than what is already done with the artifact, since these are mostly to get an idea of what the performance impact of a given PR would be. When (or if) the changes get merged, our other workflow will record the new commit on main and store it in our `uxarray-asv` repo. 

> A way to get the performance difference between two dates from the point number 2 mentioned above.

This might be a helpful reference: https://asv.readthedocs.io/en/stable/commands.html#asv-continuous